src/types/unix_time.ts

```diff
@@ -0,0 +1,27 @@
+export class UnixTime {
+	constructor(private readonly unixTime: number) {
+		if (this.unixTime < 0 || !Number.isInteger(this.unixTime) || !Number.isFinite(this.unixTime)) {
+			throw new Error('Unix time must be a positive integer!');
+		}
+	}
+
+	[Symbol.toPrimitive](hint?: string): number | string {
+		if (hint === 'string') {
+			this.toString();
+		}
+
+		return this.unixTime;
```

Comment:
For each of these `toPrimitive` we have to carefully think what are we gonna return for the `"default"` case
I think for most of the cases we are covered. Let's review this when all branches are mreged into the integration branch

Reply:
We ain't covering this case in the tests neither

Reply:
I thought about that a bit and in all those cases I think we're generally concerned with the raw underlying value. The object wrappers are just containers for code.